### ISSUE TITLE
@firebase/firestore: Firestore (9.6.4) Could not reach Cloud Firestore Backend. Backend didn't respond with 10 seconds

### ISSUE THREAD

**Issue by tb42** [firebase/firebase-js-sdk#5932]
### [REQUIRED] Describe your environment

  * Operating System version: Mac OS Monterey 12.0.1
  * Browser version: iOS simulator & Expo Go app
  * Firebase SDK version: 9.6.2
  * Firebase Product: Firestore

### [REQUIRED] Describe the problem
Any features that use the Firebase db are not rendering/are hit with this error message. The app works fine in the web browser but fails on iOS simulator and Expo Go

#### Relevant Code:

```javascript
// TODO(you): code here to reproduce the problem
```

**Comment by wu-hui** [comment]
It is likely you are seeing this issue: https://github.com/firebase/firebase-js-sdk/issues/5667

TLDR: try to disable `useFetchStreams` to see if it works for you.

**Comment by tb42** [comment]
Gave that a shot and it rendered a different issue, below is the code we're running in config (excluding credentials). There's 2 different error messages now, one in the browser and the other in iOS simulator. I've included those below as well. Suggestions? 

**Config**
`import { initializeApp } from "@firebase/app";
import { initializeFirestore} from "@firebase/firestore";
import {getStorage} from '@firebase/storage'

export const app = initializeApp(firebaseConfig);
const db = initializeFirestore(app, {useFetchStreams: false})
export const storage = getStorage(app)`

**Web error:**
<img width="780" alt="image" src="https://user-images.githubusercontent.com/16248244/151231963-a22daadf-5af2-449c-b112-32f99f7bad3d.png">

**iOS error:**
![image](https://user-images.githubusercontent.com/16248244/151232066-3a52c254-089a-4910-a936-21dc1611ac77.png)

**Comment by tb42** [comment]
Okay, I've created a private repo with my setup and sent you the invite.

**Comment by wu-hui** [comment]
Got it..I have not tried Expo before, let me give it a try today.

**Comment by tb42** [comment]
@wu-hui Any updates?

**Comment by wu-hui** [comment]
Nope, there is a more urgent issue came up that I had to deal with. 

One more thing you can try is to do the same thing with plain React Native without Expo to see if you can reproduce there, if you don't want to just wait. 

(sorry, Expo is not an officially supported platform and the turnaround time is usually long).

**Comment by wu-hui** [comment]
Hey @tb42 

I spent some time today trying to reproduce, and I could not get an Expo App running with the config you provided.

At this point, I'd have to need you to provide a complete reproduction app (plain RN preferably), with either clear instructions of steps to reproduce, and the code snippets going in the App as well.

Also, have you tried to raise an issue to Expo as well? We have a sizeable RN user base, and if Firestore is fundamentally broken we should be seeing a lot more issues by now. So maybe try some luck with Expo as well?

Thanks
